include/avm/tv.h

```diff
@@ -0,0 +1,264 @@
+   AVM_TV_RESOLUTION_720P           = 4,
+   AVM_TV_RESOLUTION_720P_3D        = 5,
+   AVM_TV_RESOLUTION_1080I          = 6,
+   AVM_TV_RESOLUTION_1080P          = 7,
+   AVM_TV_RESOLUTION_480I_PAL60     = 10,
+   AVM_TV_RESOLUTION_576P           = 11,
+   AVM_TV_RESOLUTION_720P_50HZ      = 12,
+   AVM_TV_RESOLUTION_1080I_50HZ     = 13,
+   AVM_TV_RESOLUTION_1080P_50HZ     = 14,
+} AVMTvResolution;
+
+typedef enum AVMTvVideoRegion
+{
+   AVM_TV_VIDEO_REGION_NTSC = 1,
+   AVM_TV_VIDEO_REGION_PAL,
+} AVMTvVideoRegion;
+
+/**
+ * Gets the TV video region.
+ *
+ * \return
+ * \c TRUE if the region is set to \link AVM_TV_VIDEO_REGION_NTSC NTSC \endlink.
+ * \c FALSE if the region is set to \link AVM_TV_VIDEO_REGION_PAL PAL \endlink.
+ */
+BOOL
+AVMDebugIsNTSC(void);
+
+/**
+ * Gets the TV port.
+ * 
+ * \param outPort
+ * Pointer to write the port to.
+ * 
+ * \return
+ * Always returns \c TRUE .
+ * 
+ * \sa
+ * - TVEGetCurrentPort()
+ */
+BOOL
+AVMGetCurrentPort(TVEPort *outPort);
+
+/**
+ * Gets the HDMI state.
+ * 
+ * \param outState
+ * Pointer to write the state to.
+ * 
+ * \sa
+ * - TVEGetHDMIErrorStat()
+ */
+void
+AVMGetHDMIState(TVEHdmiState *outState);
+
+/**
+ * Gets the TV aspect ratio.
+ * 
+ * \param outAspectRatio
+ * Pointer to write the aspect ratio to.
+ * 
+ * \return
+ * Always returns \c TRUE .
+ */
+BOOL
+AVMGetTVAspectRatio(AVMTvAspectRatio *outAspectRatio);
+
+/**
+ * Gets the TV resolution.
+ * 
+ * \warning
+ * If the resolution is set to #AVM_TV_RESOLUTION_480I_PAL60, this function
+ * will instead output #AVM_TV_RESOLUTION_480I.
+ * If the resolution is set to either #AVM_TV_RESOLUTION_720P_50HZ,
+ * #AVM_TV_RESOLUTION_1080I_50HZ, or #AVM_TV_RESOLUTION_1080P_50HZ, then
+ * \e nothing will be stored to \a outResolution .
+ * 
+ * \param outResolution
+ * Pointer to write the resolution to.
+ * 
+ * \return
+ * Always returns \c TRUE .
+ */
+BOOL
+AVMGetTVScanMode(AVMTvResolution *outResolution);
+
+/**
+ * Gets the TV under scan.
+ * 
+ * \param outUnderScan
+ * Pointer to write the under scan to.
+ * 
+ * \return
+ * \c 0 on success or \c -1 on failure.
+ */
+int32_t
+AVMGetTVUnderScan(uint32_t *outUnderScan);
+
+/**
+ * Determines whether a TV is currently connected and displaying the Wii U source.
+ * 
+ * Internally checks that either the #TVEHdmiState is #TVE_HDMI_STATE_DONE or
+ * the analog state is \c 1 , depending on whether the \link TVEPort port \endlink is
+ * set to \link TVE_PORT_HDMI HDMI \endlink or not.
+ * 
+ * \return
+ * \c TRUE if a TV is connected, \c FALSE otherwise.
+ * 
+ * \sa
+ * - AVMGetHDMIState()
+ * - TVEGetHDMIErrorStat()
+ * - TVEGetAnalogStat()
+ */
+BOOL
+AVMIsAVOutReady(void);
+
+/**
+ * Sets the TV aspect ratio.
+ * 
+ * \param aspectRatio
+ * Must be one of #AVMTvAspectRatio.
+ * 
+ * \return
+ * \c TRUE on success or \c FALSE on failure.
+ */
+BOOL
+AVMSetTVAspectRatio(AVMTvAspectRatio aspectRatio);
+
+/**
+ * Sets whether to show or hide the TV video out.
```

Comment:
I would probably call this "Sets whether to enable or disable the TV video out." instead.

Reply:
Reason I didn't use "enable or disable" was so this doesn't get confused with functions that make the TV report "No Signal" (I assume that's what TVEDisableVideoOut() does), since with this function the TV still receives a signal as normal and still plays audio.

Do you still want me to change it to "enable or disable" and/or add "the TV still receives a signal as normal and still plays audio"?

Maybe I could also test and add TVEDisable/EnableVideoOut() so I can link to them?

Reply:
Ah alright, that makes sense then. According to the underlying DC function is blanks the output.